### PR TITLE
executor: support single precision value in json agg

### LINKED ISSUE #37287: `JSON_ARRAYAGG` reprot unsupported second argument type
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
CREATE TABLE t1 (a FLOAT, b DOUBLE, c DECIMAL(10, 2));
INSERT INTO t1 VALUES (1.0, 2.0, 3.0),(1.0, 3.0, 9.0),(1.0, 4.0, 16.0),(1.0, 5.0, 25.0);
SELECT JSON_VALID(JSON_ARRAYAGG(a)) FROM t1;
```

### 3. What did you see instead (Required)

```
mysql> SELECT JSON_VALID(JSON_ARRAYAGG(a)) FROM t1;
ERROR 8066 (HY000): JSON_OBJECTAGG: unsupported second argument type 1
```

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?

Issue Number: close #37287 

Problem Summary:

The `json_objectagg` and `json_arrayagg` can receive float now :beers: .

### What is changed and how it works?

Convert the float value into double before writing into the map. I also moved the validation/conversion of data type from `AppendFinalResult2Chunk` to `UpdatePartialResult`, which will make it find the problem or convert the data ASAP.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

```
CREATE TABLE t1 (a FLOAT, b DOUBLE, c DECIMAL(10, 2));
INSERT INTO t1 VALUES (1.0, 2.0, 3.0),(1.0, 3.0, 9.0),(1.0, 4.0, 16.0),(1.0, 5.0, 25.0);
SELECT JSON_VALID(JSON_ARRAYAGG(a)) FROM t1;
```

It gives

```
+------------------------------+
| JSON_VALID(JSON_ARRAYAGG(a)) |
+------------------------------+
|                            1 |
+------------------------------+
```

### Release note

```release-note
Fix the issue that the single precision float cannot be used in json aggregation funtions
```

### CODE REVIEW

**Comment by xiongjiwei** [comment]
please add the manual test into the test file

File: executor/aggfuncs/func_json_objectagg.go [hunk 1]
```diff
@@ -146,7 +132,24 @@ func getRealJSONValue(value types.Datum, ft *types.FieldType) interface{} {
 		}
 	}
 
-	return realVal
+	switch x := realVal.(type) {
```
[the remark this answers is not in the document]
Comment (YangKeao) [reply executor/aggfuncs/func_json_objectagg.go:136]: Fixed.

**Review by xiongjiwei — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Pull request merged**